CMakeLists.txt

```diff
@@ -1,7 +1,13 @@
-cmake_minimum_required(VERSION 3.8)
-project(bn256 VERSION 1.0.0)
+cmake_minimum_required(VERSION 3.12)
 
-set(CMAKE_CXX_STANDARD 17)
+# set version and extract major version (currently version_major is 1).
+set(version 1.1.0)
+string(REPLACE "." ";" version_list ${version}) # transform into list
+list(GET version_list 0 version_major)
+
+project(bn256 VERSION ${version})
```

Comment:
I think you can do `project(bn256 VERSION 1.1.0)` here and then use `bn256_VERSION_MAJOR` elsewhere, instead of doing that parsing stuff.

But it seems to me like this ought to be a 2.0.0 version instead, since don't these changes break the ABI?

Reply:
> I think you can do project(bn256 VERSION 1.1.0) here and then use bn256_VERSION_MAJOR elsewhere,

yes, that works, thanks!

> But it seems to me like this ought to be a 2.0.0 version instead

Hum, I'm not sure how to tell for sure, but I'm happy to switch to 2.0.0.

Reply:
I think it's mostly academic on 1.0.0 vs 1.1.0 vs 2.0.0 for many reasons.. but setting things like
https://github.com/AntelopeIO/bn256/blob/b20f7f55ceff356eed69b8c88ebd486d713d69dc/CMakeLists.txt#L23-L25
shows an intent to version and since `std::span` is part of the public interface, switching from "not really `std::span`" to `std::span` breaks usage of the library if it had been built as a shared library and installed. Again.. that seems like a stretch of a use case.. but maybe.